mvlearn/cluster/mv_coreg_spectral.py

```diff
@@ -0,0 +1,237 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Implements multi-view spectral clustering algorithm for data with
+# multiple views.
+
+
+import numpy as np
+import scipy as sp
+from scipy.spatial.distance import cdist
+from sklearn.base import BaseEstimator
+from sklearn.cluster import KMeans
+from ..utils.utils import check_Xs
+from sklearn.exceptions import NotFittedError
+from sklearn.metrics.pairwise import rbf_kernel, polynomial_kernel
+from sklearn.neighbors import NearestNeighbors
+from .mv_spectral import MultiviewSpectralClustering
+
+AFFINITY_METRICS = ['rbf', 'nearest_neighbors', 'poly']
+
+
+class MultiviewCoRegSpectralClustering(MultiviewSpectralClustering):
+
+    r'''
+    An implementation of co-regularized multi-view spectral clustering based on
+    an unsupervied version of the co-training framework.
+    This algorithm uses the pairwise co-regularization scheme as described
+    in [#3Clu]_. This algorithm can handle 2 or more views of data.
+
+    Parameters
+    ----------
+    n_clusters : int
+        The number of clusters
+
+    random_state : int, optional, default=None
+        Determines random number generation for k-means.
+
+    info_view : int, optional, default=None
+        The most informative view. Must be between 0 and n_views-1
+        If given, then the final clustering will be performed on the
+        designated view alone. Otherwise, the algorithm will concatenate
+        across all views and cluster on the result.
+
+    max_iter : int, optional, default=10
+        The maximum number of iterations to run the clustering
+        algorithm.
+
+    n_init : int, optional, default=10
+        The number of random initializations to use for k-means clustering.
+
+    affinity : string, optional, default='rbf'
+        The affinity metric used to construct the affinity matrix. Options
+        include 'rbf' (radial basis function), 'nearest_neighbors', and
+        'poly' (polynomial)
+
+    gamma : float, optional, default=None
+        Kernel coefficient for rbf and polynomial kernels. If None then
+        gamma is computed as 1 / (2 * median(pair_wise_distances(X))^2)
+        for each data view X.
+
+    n_neighbors : int, optional, default=10
+        Only used if nearest neighbors is selected for affinity. The
+        number of neighbors to use for the nearest neighbors kernel.
+
+    v_lambda : float, optional, default=2
+        The regularization parameter. This parameter trades-off the spectral
+        clustering objectives with the degree of agreement between each pair
+        of views in the new representation. Must be a positive value.
+
```

Comment:
Can you reorder these parameters so their order matches up with the order they are in the __init__() function?

Reply:
Will do